src/main/madara-app.ts

```diff
@@ -252,3 +252,58 @@ export async function getAppSettings(appId: string) {
   }
   return {};
 }
+
+export async function fetchAllRunningApps(window: BrowserWindow) {
+  const containers = await docker.listContainers();
+  const runningApps: { [appId: string]: boolean } = {};
+
+  containers.forEach((container) => {
+    const containerName = container.Names[0].substring(1); // remove the leading "/"
+    const parts = containerName.split('-');
+    const partialAppId = parts.slice(1, 5).join('-');
+    runningApps[partialAppId] = true;
+  });
+
+  const partialAppIds = Object.keys(runningApps);
+
+  const appsFromConfigRunning = APPS_CONFIG.apps.filter((app) =>
+    partialAppIds.some((partialAppId) => app.id.startsWith(partialAppId))
+  );
+
+  appendLogs(appsFromConfigRunning as DockerAppProperties[], window);
+  return appsFromConfigRunning;
+}
+
+export async function appendLogs(
+  apps: DockerAppProperties[],
+  window: BrowserWindow
+) {
+  apps.map(async (app) => {
+    const containers = await docker.listContainers({
+      filters: {
+        name: [`${app.id}-*`],
+      },
+    });
+    await Promise.all(
+      containers.map(async (container) => {
+        const containerApp = await docker.getContainer(container.Id);
+        containerApp.attach(
```

Comment:
this code of sending logs is duplicated, can we move it to a common function to remove redundant code?

Reply:
I don't think this is fixed?

Reply:
My bad, Which is the code redundant to ?

Reply:
this one

```
container.attach(
    { stream: true, stdout: true, stderr: true },
    (err, stream) => {
      if (!stream) return;
      stream.on('data', (data) => {
        // removing the first 8 bytes as they have information about the stream
        const buffer = Buffer.from(data).subarray(8, data.length);
        const event: AppAppendLogs = {
          appId: appConfig.id,
          containerName: containerConfig.name as string,
          logs: buffer.toString(),
        };
        window.webContents.send('app-logs', event);
      });
    }
  );
```

Reply:
i fixed the redundant in the latest commit